runtime/neumann/src/lib.rs

```diff
@@ -1155,6 +1156,30 @@ impl_runtime_apis! {
 		fn generate_task_id(account_id: AccountId, provided_id: Vec<u8>) -> Hash {
 			AutomationTime::generate_task_id(account_id, provided_id)
 		}
+
+		fn calculate_optimal_autostaking(principal: i128, collator: AccountId) -> AutostakingResult {
+			let candidate_info = ParachainStaking::candidate_info(collator);
+			let money_supply = Balances::total_issuance() + Vesting::total_unvested_allocation();
+
+			let collator_stake = candidate_info.unwrap().total_counted as i128;
+			let fee = (1 * DOLLAR) as i128;
+			let duration = 90;
+			// 24 = number of collators
+			let daily_collator_rewards = (money_supply as f64 * 0.025) as i128 / 24 / 365;
```

Comment:
should it be i128 or u128?

Reply:
Is this an inadvertent truncation or is it on purpose?

Reply:
It is intentional. The formula was originally calculated with static integer values so I'm happy with this truncation.  There are many other more offensive truncations in the called method though xD